md-dao/src/main/java/com/wipro/ats/bdre/md/dao/JobDAO.java

```diff
@@ -419,14 +409,16 @@
                     initJobRowInfo.setTargetBatchId(resultBCQ.getBatchByTargetBatchId().getBatchId());
                     Criteria resultBatchCriteria = session.createCriteria(Batch.class).add(Restrictions.eq("batchId", resultBCQ.getBatchBySourceBatchId().getBatchId()));
                     Batch resultBatch = (Batch) resultBatchCriteria.list().get(0);
-                    try {
+                    if(resultBatch!=null &&resultBatch.getInstanceExec()!=null) {
                         initJobRowInfo.setSourceInstanceExecId(resultBatch.getInstanceExec().getInstanceExecId());
-                    } catch (NullPointerException e) {
+                    }
+                    else
+                    {
                         LOGGER.info("source Instance Exec Id is  NULL");
                         initJobRowInfo.setSourceInstanceExecId(null);
                     }
                     Criteria fileBatchCriteria = session.createCriteria(File.class).add(Restrictions.eq("id.batchId", resultBCQ.getBatchBySourceBatchId().getBatchId()));
-                    if (fileBatchCriteria.list().size() > 0) {
+                    if (!fileBatchCriteria.list().isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)